services/proxy/pkg/config/config.go

```diff
@@ -78,11 +79,23 @@ var (
 	RouteTypes = []RouteType{QueryRoute, RegexRoute, PrefixRoute}
 )
 
-// AuthMiddleware configures the proxy http auth middleware.
-type AuthMiddleware struct {
+// AuthenticationMiddleware configures the proxy http authentication middleware.
+type AuthenticationMiddleware struct {
 	CredentialsByUserAgent map[string]string `yaml:"credentials_by_user_agent"`
 }
 
+// AuthorizationMiddleware configures the proxy http authorization middleware.
+type AuthorizationMiddleware struct {
+	OPA AuthorizationMiddlewareOPA `yaml:"opa"`
+}
+
+// AuthorizationMiddlewareOPA configures the opa policy middleware.
+type AuthorizationMiddlewareOPA struct {
+	Enabled  bool     `yaml:"enabled"`
+	Policies []string `yaml:"policies"`
+	Timeout  int      `yaml:"timeout"`
```

Comment:
Better to use `time.Duration` as  type. That way users can define `10s` or `1m` and do not need to check if it is seconds or milliseconds

Reply:
good point, ❤️ Done